cmd/collector/main.go

```diff
@@ -75,6 +76,17 @@ func main() {
 		logger.Fatal("Error while searching for management config", zap.Error(err))
 	}
 
+	// Change working directory to executable directory (for Windows services)
+	ex, err := os.Executable()
+	if err != nil {
+		logger.Warn("Failed to retrieve executable directory", zap.Error(err))
+	} else {
+		execDirPath := filepath.Dir(ex)
+		if err := os.Chdir(execDirPath); err != nil {
+			logger.Warn("Failed to modify current working directory", zap.Error(err))
+		}
+	}
```

Comment:
I think this could lead to some surprising behavior when running as not-a-service, or modifying the CWD for services on linux (which isn't often, but it could happen).

Can we make this logic only run for windows as a service? Possibly move it into internal/service/service_windows.go, before svc.Run is called.

Reply:
@BinaryFissionGames can you give me a more specific scenario that you are worried about? either way, you're right that it's probably not a bad idea to keep the change a little more compartmentalized

Reply:
It's very edge case, but I'm thinking of a case where we run the collector from a different directory than where the exe is.

e.g. I have a folder on my system where I keep some sample configs, and run manually /opt/observiq-otel-collector/observiq-otel-collector --config ./test-config.yaml

It'd be very surprising to me that ./test-config.yaml does not refer to the directory relative to my CWD, but relative to the binary. Could see it being something the field team could possibly run into while doing support.

Reply:
@BinaryFissionGames yeah that's totally a fair and reasonable worry. I made the changes you requested and will test them out.

Reply:
@BinaryFissionGames alright made the changes and tested them. great catch!